### PR TITLE
Property tests for Storage trait impl

### LINKED ISSUE #730: Test Storage impl for ConsensusState
As our lengthy investigation with https://github.com/qdrant/qdrant/pull/710 shows - we need to carefully check all the invariants by ourselves in our implementation. 

It is suggested to:
1. Abstract ToC functions needed for ConsensusState with a trait
2. Test Storage impl for ConsensusState (as close as possible to raft MemStorage tests https://github.com/tikv/raft-rs/blob/master/src/storage.rs#L523) 

We might use property testing libraries for better coverage - https://crates.io/crates/proptest

### PR DESCRIPTION
**Pull request by e-ivkov**
Closes #730 

`Storage::initial_state` and `Storage::snapshot` are not tested as they are not good targets for property testing and are rather straightforward.

### CODE REVIEW

File: lib/storage/src/content_manager/consensus_state.rs [hunk 1]
```diff
@@ -917,4 +931,47 @@ mod tests {
...
+            &self,
+            _data: super::CollectionsSnapshot,
+        ) -> Result<(), crate::content_manager::errors::StorageError> {
+            Ok(())
+        }
+    }
+
+    prop_compose! {
+        fn gen_entries(max_entries: u64)(n in 0u64..max_entries) -> Vec<Entry> {
+            (1..(n+1)).into_iter().map(|index| Entry {index, term: 1, ..Default::default()}).collect::<Vec<Entry>>()
```
Comment (agourlay) [inline lib/storage/src/content_manager/consensus_state.rs:959]: Is it possible to increment the term from time to time as well?
Could be using a randomized approach or just modding around the index.

File: lib/storage/src/content_manager/consensus_state.rs [hunk 2]
```diff
@@ -409,6 +416,17 @@ impl Storage for ConsensusState {
         _context: GetEntriesContext,
     ) -> raft::Result<Vec<RaftEntry>> {
         let max_size: Option<_> = max_size.into();
+        if low < self.first_index()? {
```
Comment (agourlay) [inline lib/storage/src/content_manager/consensus_state.rs:419]: is this a bug fix?

File: lib/storage/src/content_manager/consensus_state.rs [hunk 3]
```diff
@@ -917,4 +942,89 @@ mod tests {
         // Even when `max_size` is `0` this fn should return at least 1 entry
         assert_eq!(wal.entries(4, 5, Some(0)).unwrap().len(), 1)
     }
+
+    struct NoCollections;
```
Comment (agourlay) [inline lib/storage/src/content_manager/consensus_state.rs:946]: Where is this struct used?

File: lib/storage/src/content_manager/mod.rs [hunk 4]
```diff
@@ -28,3 +33,14 @@ pub mod consensus_ops {
         }
     }
 }
+
```
Comment (agourlay) [inline lib/storage/src/content_manager/mod.rs:36]: Can you please put some basic Rust doc on this trait.

File: lib/storage/src/content_manager/consensus_state.rs (hunk 3 above)
Comment (agourlay) [reply lib/storage/src/content_manager/consensus_state.rs:946]: I must have been tired when writing this comment, I clearly see it used in `setup_storages` now :sleepy:

Comment (e-ivkov) [reply lib/storage/src/content_manager/consensus_state.rs:946]: No worries!

**Pull request merged**